client/extensions/woocommerce/woocommerce-services/views/shipping-label/label-purchase-modal/purchase-button.js

```diff
@@ -0,0 +1,107 @@
+/** @format */
+
+/**
+ * External dependencies
+ */
+import React from 'react';
+import PropTypes from 'prop-types';
+import { connect } from 'react-redux';
+import { bindActionCreators } from 'redux';
+import { localize } from 'i18n-calypso';
+
+/**
+ * Internal dependencies
+ */
+import Button from 'components/button';
+import getPDFSupport from 'woocommerce/woocommerce-services/lib/utils/pdf-support';
+import formatCurrency from 'lib/format-currency';
+import {
+	confirmPrintLabel,
+	purchaseLabel,
+} from 'woocommerce/woocommerce-services/state/shipping-label/actions';
+import {
+	getShippingLabel,
+	isLoaded,
+	getTotalPriceBreakdown,
+	canPurchase,
+} from 'woocommerce/woocommerce-services/state/shipping-label/selectors';
+
+class PurchaseButton extends React.Component {
+	static propTypes = {
+		siteId: PropTypes.number.isRequired,
+		orderId: PropTypes.number.isRequired,
+	};
+
+	getPurchaseButtonLabel = () => {
+		const { form, ratesTotal, translate } = this.props;
+
+		if ( form.needsPrintConfirmation ) {
+			return translate( 'Print' );
+		}
+
+		if ( form.isSubmitting ) {
+			return translate( 'Purchasing…' );
+		}
+
+		const noNativePDFSupport = 'addon' === getPDFSupport();
+
+		if ( this.props.canPurchase ) {
+			if ( noNativePDFSupport ) {
+				return translate( 'Buy (%s)', { args: [ formatCurrency( ratesTotal, 'USD' ) ] } );
+			}
+
+			return translate( 'Buy & Print (%s)', { args: [ formatCurrency( ratesTotal, 'USD' ) ] } );
+		}
+
+		if ( noNativePDFSupport ) {
+			return translate( 'Buy' );
+		}
+
+		return translate( 'Buy & Print' );
+	};
+
+	getPurchaseButtonAction = () => {
+		const { form, orderId, siteId } = this.props;
+		if ( form.needsPrintConfirmation ) {
+			return () => this.props.confirmPrintLabel( orderId, siteId );
+		}
+		return () => this.props.purchaseLabel( orderId, siteId );
```

Comment:
Nitpick: I prefer building these functions in the `mapDispatchToProps`, so the component only needs to call `this.props.purchaseLabel()` without caring for the `orderId` or `siteId`. Every change that makes the component more "dumb" is good IMHO :)